src/Task.cc

```diff
@@ -1221,14 +1204,43 @@ void Task::emulate_syscall_entry(const Registers& regs) {
   set_regs(r);
 }
 
-void Task::did_waitpid(WaitStatus status, siginfo_t* override_siginfo) {
+void Task::did_waitpid(WaitStatus status) {
   Ticks more_ticks = hpc.read_ticks();
   // Stop PerfCounters ASAP to reduce the possibility that due to bugs or
   // whatever they pick up something spurious later.
   hpc.stop();
   ticks += more_ticks;
   session().accumulate_ticks_processed(more_ticks);
 
+  // After PTRACE_INTERRUPT, any next two stops may be a group stop caused by
+  // that PTRACE_INTERRUPT (or neither may be). This is because PTRACE_INTERRUPT
+  // generally lets other stops win (and thus doesn't inject it's own stop), but
+  // if the other stop was already done processing, even we didn't see it yet,
+  // the stop will still be queued, so we could see the other stop and then the
+  // PTRACE_INTERRUPT group stop.
+  // When we issue PTRACE_INTERRUPT, we this set this counter to 2, and here
+  // we decrement it on every stop such that while this counter is positive,
+  // any group-stop could be one induced by PTRACE_INTERRUPT
+  bool siginfo_overriden;
```

Comment:
Oops, sorry - languages where this can happen are not my bread and butter anymore, so I keep making these mistakes ;).